src/components/store/SizeSelector/index.tsx

```diff
@@ -0,0 +1,56 @@
+import styles from '@/components/store/SizeSelector/style.module.scss';
+import { PublicMerchItemOption } from '@/lib/types/apiResponses';
+import { useEffect, useId, useState } from 'react';
+
+interface SizeSelectorProps {
+  currSize: string | undefined;
+  options: PublicMerchItemOption[];
+  // Justification for disabling rules: This seems to be a false positive.
+  // https://stackoverflow.com/q/63767199/
+  // eslint-disable-next-line no-unused-vars
+  setSize: (value: string) => void;
+}
+
+const SizeSelector = ({ currSize, options, setSize }: SizeSelectorProps) => {
+  const [mounted, setMounted] = useState(false);
+  const myID = `${useId()}`;
+
+  useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    return null;
+  }
+
+  const myOptions = options.map(val => {
+    const tempId = val.metadata?.value ? `${val.metadata.value}${myID}` : `${val.uuid}${myID}`;
+    return (
+      <>
```

Comment:
Hmm, does React not give a warning about this missing a `key` prop? React usually requires a `key` prop when you return a list of elements, like here, so it can keep track of which is which

You can't add a prop directly to `<></>` though. You'd have to import `Fragment` (which is what `<>` is short for) from React then do

```suggestion
      <Fragment key={val.uuid}>
```

Reply:
<img width="363" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/b1725390-d690-4eac-9516-8d98fa258c81">

Reply:
React was not giving me an error for this, which is odd.

Reply:
you only need it if you're returning an array of elements. for the `Fragment` in your screenshot, it's not in an array, so you don't need a `key`

Reply:
Which lines were you referring to then? I think I'm misunderstanding.

Reply:
the lines inside `const myOptions`